### PR TITLE
Feature - Create calendar screen page

### PR DESCRIPTION
**Pull request by at-hungnguyen3**

### CODE REVIEW

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Date;
+
+public class CalendarPage extends BasePage {
+    JavascriptExecutor jsExecutor;
+
+    @AndroidFindBy(className = "android.widget.ImageButton")
+    @iOSFindBy(className = "UIAButton")
+    private MobileElement back;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:27]: btnBack

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    @AndroidFindBy(id = "action_bar")
+    @iOSFindBy(className = "UIANavigationBar")
+    protected MobileElement actionBar;
+
+    @AndroidFindBy(xpath = "//*[@text='History']")
+    @iOSFindBy(id = "History")
+    private MobileElement history;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:35]: tvHistory

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    @AndroidFindBy(xpath = "//*[@text='Some servings']")
+    @iOSFindBy(xpath = "//*[@text='Some servings']")
+    private MobileElement someServings;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:39]: tvSomeServings

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    @AndroidFindBy(xpath = "//*[@text='All servings']")
+    @iOSFindBy(xpath = "//*[@text='All servings']")
+    private MobileElement allServings;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:43]: tvAllServings

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPageAndroid.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        waitForElementDisplay(time);
+        return time.getText();
+    }
+
+    public boolean isCurrentTime() {
+        String[] parts = currentTime.split("(?=-)");
+        String month = parts[1];
+        String year = parts[2];
+        if ((getTimeText().contains(month)) && (getTimeText().contains(year))) {
+            return true;
```
Comment (at-anhquach) [inline Training/src/test/java/page/exam/at_hungnguyen3/CalendarPageAndroid.java:54]: return (getTimeText().contains(month)) && (getTimeText().contains(year))

File: Training/src/test/java/stepdefs/exam/CalendarDefinitions.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        });
+        When("^I tap on next button$", () -> {
+            calendarPageAndroid.clickToNextButton();
+        });
+        When("^I tap on next previous button$", () -> {
+            calendarPageAndroid.clickToPreviousButton();
+        });
+
+    }
+}
```
Comment (at-anhquach) [inline Training/src/test/java/stepdefs/exam/CalendarDefinitions.java:52]: enter line

**Review by at-anhquach — APPROVED** [review]

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java (hunk 1 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:27]: done

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java (hunk 2 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:35]: done

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java (hunk 3 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:39]: done

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java (hunk 4 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/page/exam/at_hungnguyen3/CalendarPage.java:43]: done

File: Training/src/test/java/page/exam/at_hungnguyen3/CalendarPageAndroid.java (hunk 5 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/page/exam/at_hungnguyen3/CalendarPageAndroid.java:54]: done

File: Training/src/test/java/stepdefs/exam/CalendarDefinitions.java (hunk 6 above)
Comment (at-hungnguyen3) [reply Training/src/test/java/stepdefs/exam/CalendarDefinitions.java:52]: done

**Pull request merged**